src/main/java/com/ernie/TicketApp/repository/UserDAO.java

```diff
@@ -93,4 +95,46 @@ public List<User> getAllUsers() {
         return allUsers;
     }
 
+    @Transactional
+    public void updateUserAddTicket(UUID id) {
```

Comment:
the core idea of p.6 from HW11 is correct, but the implementation - not. Follow next instructions:
1. when you work with  `try (PreparedStatement preparedStatement = dataSource.getConnection().prepareStatement(...)) {...}` then:

-  you are responsible for all interactions with database, while the idea is to transfer this responsibility to Spring.
- each call of `executeUpdate()` method will make changes in database independently, while the idea is to perform them in 1 transaction. Both of 2nd and 3d changes have to be performed or rolled back

In order to fix it, you can:
- work with `JdbcTemplate` + `@Transactional` instead of `try (PreparedStatement preparedStatement = dataSource.getConnection().prepareStatement(...)) {...}` :
add this bean in configuration class:
```java
    @Bean
    public JdbcTemplate jdbcTemplate(DataSource dataSource) {
        return new JdbcTemplate(dataSource);
    }
```
and to `UserDAO` class:
```java
    private final JdbcTemplate jdbcTemplate;
    public UserDAO(DataSource dataSource, JdbcTemplate jdbcTemplate) {
        this.dataSource = dataSource;
        this.jdbcTemplate = jdbcTemplate;
    }
```
- call 2 queries on this object:
```java
jdbcTemplate.update("your query", newTicket.getId().toString(), user.getId().toString(), newTicket.getTicketType().name(), newTicket.getCreationDateTime().toString());
jdbcTemplate.update("your query", yourVariables);
```